File: Easy/1-two-sum/solution.rs
//leetcode.com/problems/two-sum/
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut s: HashMap<i32, i32> = HashMap::new();
        for (index, num) in nums.iter().enumerate() {
            let val = target - *num;
            if s.contains_key(&val) {
                return vec![s[&val], index as i32];
            }
            s.insert(*num, index as i32);
        }
        Vec::new()
    }
}

File: easy/1-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> m;
        vector<int> res;
        for(int i{0};i<nums.size();++i)
        {
            auto it=m.find(target-nums[i]);
            if(it!=nullptr)
            {
                res.push_back(i);
                res.push_back(it->second);
                return res;
            }
            m[nums[i]]=i;
        }
        return res;
    }
};
